src/rgw/rgw_rest_s3.cc

```diff
@@ -3229,6 +3229,8 @@ int RGW_Auth_S3::authorize(RGWRados *store, struct req_state *s)
     return 0;
   }
 
+  s->aws4_auth_presigned_url_bugfix = false;
+
   if (!s->http_auth || !(*s->http_auth)) {
```

Comment:
Having a look at:

https://github.com/jmunhoz/ceph/blob/8f915e070a5f7847e785f556e89bda09540d5321/src/rgw/rgw_rest_s3.cc#L3679

you'll see it is guarded by 'using_qs'

<pre>
if (!s->aws4_auth_presigned_url_bugfix && using_qs && (token == "host")) {
</pre>

In the case of presigned urls using_qs is true:

https://github.com/jmunhoz/ceph/blob/8f915e070a5f7847e785f556e89bda09540d5321/src/rgw/rgw_rest_s3.cc#L3425

and it is guarded by:

<pre>
if ((!s->http_auth) || !(*s->http_auth)) {
</pre>

This is the same condition guarding the first conditional branch so I think we don't need add the second authorize_v4 call cause it will not be executed due to using_qs.